src/tests/quantuminspire/test_api.py

```diff
@@ -1126,6 +1142,14 @@ def __mocks_for_api_execution(self):
 
         return expected_job_result, job_mock, asset_mock, backend_mock, project_mock
 
+    def test_execute_qasm_cancelled_job(self):
+        mocks = self.__mocks_for_api_execution(fake_no_results=True)
+        api = QuantumInspireAPI('FakeURL', self.authentication, coreapi_client_class=self.coreapi_client)
+        qasm = 'version 1.0...'
+        number_of_shots = 1024
+        results = api.execute_qasm(qasm, number_of_shots=number_of_shots, backend_type=1)
+        self.assertEqual(results, OrderedDict())
```

Comment:
How does the user know if the job was cancelled, failed, or just timed out?

Reply:
When the result was False, now a minimal result object is created with an empty histogram and an error message. so in all cases some result is sent back and the user can check the result for more information.